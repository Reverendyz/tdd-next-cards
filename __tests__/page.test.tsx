import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Page from '../app/page'
 
describe('Main page tests', () => {
  beforeAll(() =>{
    render(<Page/>)
  })
  
  it('renders two divs within the nav element', () => {
    const main = screen.getByRole("main-text")
    expect(main).toBeInTheDocument()
  })

})

test('renders Page and saves snapshot', () => {
  const { asFragment } = render(<Page />);
  expect(asFragment()).toMatchSnapshot();
});