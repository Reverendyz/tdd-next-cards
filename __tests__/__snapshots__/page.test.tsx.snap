// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Page and saves snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]"
    id="home"
  >
    <div
      class="text-4xl"
      role="main-text"
    >
      Get 
      <code
        class="transition ease-in-out delay-150 bg-blue-500 hover:-translate-y-1 hover:scale-110 hover:bg-indigo-500 duration-300 hover:bg-gradient-to-r from-purple-500 hover:to-blue-1000 rounded-md"
      >
        ...more
      </code>
       knowledge
    </div>
  </div>
</DocumentFragment>
`;
